### PR TITLE
Accounts API v1

### PR DESCRIPTION
**Pull request by nicola**
List of APIs we are planning on implementing

#### LDNODE
- [x] `/api/accounts/signin`: 302 to oidc if possible
- [x] `/api/accounts/signout`: removes the session
- [x] `/api/accounts/recover`: generates a token and contacts the user sending the token
- [x] `/api/accounts/validateToken`: user can claim a token, gets the cookies and gets redirected to /
- [x] `/api/accounts/new`: creates a new account on LDP and create it under OIDC https://github.com/solid/node-solid-server/pull/349

#### OIDC integration
- [x] `/api/accounts/signin`
- [x] `/api/accounts/new`: call to OIDC to create new user https://github.com/solid/node-solid-server/pull/349

for v2 see https://github.com/solid/node-solid-server/issues/350

## Other things to do:

- [x] move all the endpoints to `/api`
- [ ] get OIDC finalized https://github.com/solid/node-solid-server/pull/330

### CODE REVIEW

File: lib/api/accounts/signin.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+function signin () {
+  return (req, res, next) => {
+    if (!validUrl.isUri(req.body.webid)) {
+      return res.status(406).send('This is not a valid URI')
+    }
+
+    request({ method: 'OPTIONS', uri: req.body.webid }, function (err, req) {
+      if (err) {
+        res.status(406).send('Did not find a valid endpoint')
```
Comment (dmitrizagidulin) [inline lib/api/accounts/signin.js]: Why 406 here? I thought that error only dealt with `Accept` headers & content types?

Comment (nicola) [inline lib/api/accounts/signin.js]: if you send a bad URI, you should know about it

Comment (dmitrizagidulin) [inline lib/api/accounts/signin.js]: I agree. But this should be a 400 error, not 406.

Comment (nicola) [inline lib/api/accounts/signin.js]: even for the 409 happening later?

**Pull request merged**